backend/app/assets/javascripts/spree/backend/views/order/shipping_method.js

```diff
@@ -0,0 +1,52 @@
+Spree.Views.Order.ShippingMethod = Backbone.View.extend({
+  tagName: 'tr',
+  className: 'edit-shipping-method',
+
+  events: {
+    "click .js-edit":   "onEdit",
+    "click .js-save":   "onSave",
+    "click .js-cancel": "onCancel",
+    "change select": "onChange",
+  },
+
+  initialize: function(options) {
+    this.shippingRateId = this.model.get('selected_shipping_rate').get('id')
+    this.render();
+  },
+
+  onEdit: function(event) {
+    this.editing = true;
+    this.render();
+  },
+
+  onSave: function(event) {
+    this.editing = false;
+    this.model.save({
+      selected_shipping_rate_id: this.$('select').val()
+    }, {
+      patch: true,
+      success: function() {
+        // FIXME: should update page without reloading
+        window.location.reload();
```

Comment:
Changing the shipping method may change the amount of the order, so the sidebar needs updating. This will eventually be done without a reload, but is outside the scope of this PR (which is the first step towards removing it).

This doesn't introduce the reload, it was moved from [here](https://github.com/jhawthorn/solidus/blob/12565691ca470afd93cc577e3908f1551a8248f1/backend/app/assets/javascripts/spree/backend/shipments.js#L310)

Reply:
Thanks for clarification